### DISCUSSION TITLE
Fast data upload speed (Nano BLE Sense / ESP-01s)

### DISCUSSION THREAD

**Discussion by JamieMcQuire23** [JAndrassy/WiFiEspAT#59] [Q&A]
Hi,

 Fantastic library! Using this in a project atm. I'm looking to use the ESP-01s with the Nano 33 BLE Sense connected via hardware serial (Serial1) , to send binary data from an SD card to a web server. I'm finding that at 115200 Baud I can achieve approximately 3kbps data rates which is too slow for my application. Does anyone have any strategies for improving the data rate?

**Comment by JAndrassy** [comment]
on Mega 500000 baud worked. I don't remember if I did a speed test with Nano BLE.

you can use the Tools/ChangeATBaudRate 'example' to set the baud rate to AT firmware

**Reply by JamieMcQuire23** [reply]
Set the Baud at 460800 with a slow data rate. Does it have to be 500000?

**Reply by JAndrassy** [reply]
for Mega at 16 MHz 500000 is an optimal value because of clock signal division

**Reply by JamieMcQuire23** [reply]
Seemed to be an issue with setting 500,000 on the Nano BLE. Not sure why but think you need to use set baud rates (got these from a data sheet). Having issue with the client.write(file) function not actually sending data but the time looks like it's around 25kbps. Any suggestions why it might not be working? I'm using SDFat and not SD think that's the only difference with my code.
